addons/test_website/static/tests/tours/website_page_properties.js

```diff
@@ -0,0 +1,395 @@
+import {
+    assertPathName,
+    clickOnSave,
+    getClientActionUrl,
+    registerWebsitePreviewTour,
+} from "@website/js/tours/tour_utils";
+import { waitFor } from "@odoo/hoot-dom";
+
+const openPagePropertiesDialog = [
+    {
+        content: "Open Site backend menu",
+        trigger: ':visible [data-menu-xmlid="website.menu_site"]',
+        run: "click",
+    },
+    {
+        content: "Open page properties dialog",
+        trigger: ':visible [data-menu-xmlid="website.menu_page_properties"]',
+        run: "click",
+    },
+];
+
+const clickOnSaveButtonStep = {
+    content: "Click on Save & Close",
+    trigger: ".o_form_button_save:enabled",
+    run: "click",
+};
+
+const openCreatePageDialog = [
+    {
+        content: "Open create content menu",
+        trigger: ".o_new_content_container a",
+        run: "click",
+    },
+    {
+        content: "Create a new page",
+        trigger: 'a[title="New Page"]',
+        run: "click",
+    },
+];
+
+function goToUrl(url, edition = false) {
+    return [
+        {
+            content: `Go to ${url}`,
+            trigger: "body",
+            run() {
+                window.location = getClientActionUrl(url, edition);
+            },
+        },
+    ];
+}
+
+/**
+ * @param {string} selector
+ * @param {string} value
+ * @return {Array<TourStep>}
+ */
+function edit(selector, value) {
+    return [
+        {
+            content: `Set ${selector} to ${value}`,
+            trigger: selector,
+            run: `edit ${value}`,
+        },
+        {
+            content: "Dispatch change event on field",
+            trigger: selector,
+            run() {
+                this.anchor.dispatchEvent(new Event("change"));
+            },
+        },
+    ];
+}
+
+/**
+ * For when tour utils doesn't detect the DOM changes...
+ * Seems to happen when watching for an element in the page template selection
+ * modal that doesn't exist yet, then appears. I suspect the DOM changes to the
+ * modal don't trigger a new search of the `trigger`.
+ */
+function waitForSelector(selector) {
+    return [
+        {
+            content: `Wait for ${selector}`,
+            trigger: "body",
+            async run() {
+                return waitFor(selector, {
+                    timeout: 5000,
+                });
+            },
+        },
+    ];
+}
```

Comment:
@pipu

Reply:
Currently, the macro.js system searches DIRECTLY in the DOM if it finds the element. If it does, it goes to the next step and searches for it directly as well and so on. If the trigger is not specific enough, the trick can take a wrong path. We are currently looking to fix this behavior. The way to do it here is actually not the "good" behavior because the waitFor waits for the first element that will match with the selector which is exactly the same as what I explain. It is better to use a setTimeout with a promise and really wait a few milliseconds before searching in the DOM so that the mutations take place before searching for the element.

Reply:
`return [
        {
            content: `Wait for ${selector}`,
            trigger: "body",
            async run() {
                return new Promise((r) => setTimeout(r, 300));
            },
        },
        {
            content: `Find ${selector}`,
            trigger: selector,
        },
    ];`

Reply:
@pipu-odoo Sorry, I did not understand. The tour system is listening to mutations and checking the next trigger after each mutation, right? It obviously cannot work without that, searching the next trigger in the DOM directly is often ok but very often not, right? (e.g. click on the switch-to-kanban button, the kanban takes time to appear => the next trigger "checking a kanban card" does not have to first have a step saying "wait for 300ms").

Can you clarify?